zipline/data/us_equity_pricing.py

```diff
@@ -710,17 +715,8 @@ class PanelDailyBarReader(DailyBarReader):
     first_trading_day : pd.Timestamp
         The first trading day in the dataset.
     """
+    @preprocess(panel=call(verify_indices_all_unique))
```

Comment:
I like `preprocess`.  It's a little magical, but it helps a lot for uncluttering functions, and I think it provides a nice declarative view of a function's preconditions.  I could see an argument for leaving the call inline here though.